dxp/build_process.md

```diff
@@ -0,0 +1,67 @@
+
+### Build Pipeline
+
+Following the sequence we can describe in more detail what happens in each phase briefly.
+
+By default the CLI expects its source code to be in `src/main/resources/META-INF/resources` and is compiled into the `build/node/packageRunBuild/resources` folder.
+
+#### 1. Setup env
+
+Only set `NODE_ENV` to `production`, you can easily overwrite it just by setting `env NODE_ENV=development liferay-npm-scripts build`.
+
+#### 2. Validate `npmscripts.config.js`
+
+The CLI will validate the `npmscripts.config.js` configuration file looking for required keys, just to avoid errors that are difficult to track later.
+
+> You can read more about the [npmscripts.config.js settings](https://github.com/liferay/liferay-npm-tools/tree/master/packages/liferay-npm-scripts#config) in the [liferay-npm-tools](https://github.com/liferay/liferay-npm-tools) repository.
+
+#### 3. Soy
+
+If there is a `.soy` file in the module, `liferay-npm-scripts` will build Soy and also Babel it next and drop it into the build destination folder. Unlike the other phases, before executing the babel the Soy build will compile your `.soy` code and drop it into a temporary folder (by default in `build/npmscripts`) until the babel can transform and send it to destination folder.
+
+The CLI will run the [`metalsoy`](https://github.com/metal/metal-tools-soy/tree/4.x) tool to handle with the Soy files.
+
+#### 4. Babel
+
+As part of the build process, [babel](https://babeljs.io/) is used for transformation, for React with JSX syntax for example and to allow the use of new JavaScript features.
+
+#### 5. Webpack
+
+The CLI can also run the [webpack](https://webpack.js.org/) as part of the process if there is a [`webpack.config.js`](https://webpack.js.org/configuration/#options) configuration file.
+
+> The CLI will also merge the babel config with the `webpack.config.js` configuration if the loader `babel-loader` is present in the configuration.
+
+#### 6. Liferay npm bundler
+
+Running [liferay-npm-bundler](https://github.com/liferay/liferay-js-toolkit/tree/master/packages/liferay-npm-bundler) is super important to allow your project's dependencies and source code to be converted from ES5 to AMD and can be deployed as an OSGI package later. So, Liferay Portal will be able to load the files of your module and its dependencies when requested.
+
+> The [liferay-js-toolkit](https://github.com/liferay/liferay-js-toolkit) repository wiki is full of documents about how [liferay-npm-bundler](https://github.com/liferay/liferay-js-toolkit/wiki) works, check it out for more information.
```

Comment:
The "JS Toolkit wiki"....

![chaos](https://media.giphy.com/media/joV1k1sNOT5xC/giphy.gif)